src/mod/MEs_Merge_FN.c

```diff
@@ -55,7 +56,20 @@ MEdge_ptr MEs_Merge_FN(MEdge_ptr e1, MEdge_ptr e2, int topoflag) {
     List_Delete(efaces2);
   }
 
-  ME_Delete(e2,0);
+  nsets = MESH_Num_MSets(mesh);
```

Comment:
I wonder whether this can be done more efficiently - at a higher level we keep a note of edges, vertices etc that have been deleted and update the sets once. This seems like it might slow it down quite a bit.